InnerEye/ML/SSL/lightning_containers/ssl_image_classifier.py

```diff
@@ -0,0 +1,68 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+from pathlib import Path
+from typing import Any, Dict
+
+import param
+
+from InnerEye.ML.SSL.datamodules_and_datasets.datamodules import InnerEyeVisionDataModule
+from InnerEye.ML.SSL.lightning_containers.ssl_container import InnerEyeDataModuleTypes, SSLContainer
+from InnerEye.ML.SSL.utils import create_ssl_image_classifier
+from InnerEye.ML.lightning_container import LightningModuleWithOptimizer
+
+
+class SSLClassifierContainer(SSLContainer):
+    """
+    This module is used to train a linear classifier on top of a frozen (or not) encoder.
+
+    If you are running on AML, you can specify the SSL training run id via the --pretraining_run_recovery_id flag. This
+    will automatically download the checkpoints for you and take the latest one as the starting weights of your
+    classifier.
+
+    If you are running locally, you can specify the path to your SSL weights via the --local_ssl_weights_path parameter.
+
+    See docs/self_supervised_models.md for more details.
+    """
+    freeze_encoder = param.Boolean(default=True, doc="Whether to freeze the pretrained encoder or not.")
+    local_ssl_weights_path = param.ClassSelector(class_=Path, default=None, doc="Local path to SSL weights")
+
+    def create_model(self) -> LightningModuleWithOptimizer:
+        """
+        This method must create the actual Lightning model that will be trained.
+        """
+        if self.local_ssl_weights_path is None:
+            assert self.extra_downloaded_run_id is not None
+            try:
+                path_to_checkpoint = self.extra_downloaded_run_id.get_best_checkpoint_paths()
+            except FileNotFoundError:
+                path_to_checkpoint = self.extra_downloaded_run_id.get_recovery_checkpoint_paths()
```

Comment:
do you mind if we add a short logging.info here saying sth like
"best checkpoint is not found instead found the latest one"